pkgs/applications/virtualization/singularity/generic.nix

```diff
@@ -0,0 +1,199 @@
+{ lib
+, buildGoModule
+, cryptsetup
+, makeWrapper
+, util-linux
+, which
+, bash
+, coreutils
+, go
+, gpgme
+, libuuid
+  # This is for nvidia-container-cli
+, nvidia-docker
+, openssl
+, squashfsTools
+  # Configurations
+, pname
+, version
+, src
+, projectName # "apptainer" or "singularity"
+  # These cannot be correctly override with `overrideAttrs`
+, vendorSha256 ? null
+, deleteVendor ? false
+, proxyVendor ? false
+, doCheck ? true
+, enableNvidiaContainerCli ? true
+  # This is required to run containers without suid
+, withoutSuid ? true
+, starterSuidPath ? null
+  # Remove the symlink to `singularity` when projectName != "singularity"
+, removeCompat ? true
+, extraDescription ? ""
+, extraMeta ? { }
+}:
+
+buildGoModule {
+  inherit pname version src;
+
+  # All three projects provide vendored source tarball
+  # so we set vendorSha256 = null;
+  # One who use unvendored source can override this
+  # with `override`
+  inherit vendorSha256 deleteVendor proxyVendor;
```

Comment:
I would rather remove this options and make the wrapper separate so that people can build their own binary with buildGoModule from scratch and pass it into the wrapper.